src/radio/radio-device-manager.ts

```diff
@@ -70,6 +70,35 @@ export class RadioDeviceManager implements smarthome.DeviceManager {
     );
   }
 
+  /**
+   * Fulfills a TcpRequestData.
+   * @param tcpRequestData  The TcpRequestData to source radio parameters from.
+   * @returns  A promise that resolves to the determined TcpResponseData.
+   */
+  private async processTcpRequestData(
+    tcpRequestData: smarthome.DataFlow.TcpRequestData
+  ): Promise<smarthome.DataFlow.TcpResponseData> {
+    const localAddress = this.deviceIdToAddress.get(tcpRequestData.deviceId);
+    const requestId = tcpRequestData.requestId;
+    const deviceId = tcpRequestData.deviceId;
+    const port = tcpRequestData.port;
+
+    if (tcpRequestData.operation === smarthome.Constants.TcpOperation.READ) {
+      const tcpResponse = await this.radioController.readTcpSocket(
```

Comment:
Can the readTcpSocket not return a TcpResponseData?

Reply:
If I'm not mistaken, if `readTcpSocket()` returned a `TcpResponseData`, I would have to either pass the `deviceId` and `requestId` to `readTcpSocket()`, or allow `requestId` and `deviceId` be undefined in `TcpResponseData` and set them after.  I think it makes sense as-is.